samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb

```diff
@@ -0,0 +1,1434 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then implement a custom `Handler`.\n",
+    "\n",
+    "A [`Handler`](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/handler.py) must implement the following interface."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Handler:\n",
+    "    \"\"\"Interface for Handler class to handle prediction requests.\"\"\"\n",
+    "\n",
+    "    def __init__(\n",
+    "        self, gcs_artifacts_uri: str, predictor: Optional[Type[Predictor]] = None,\n",
+    "    ):\n",
+    "        \"\"\"Initializes a Handler instance.\n",
+    "        Args:\n",
+    "            gcs_artifacts_uri (str):\n",
+    "                Required. The value of the environment variable AIP_STORAGE_URI.\n",
+    "            predictor (Type[Predictor]):\n",
+    "                Optional. The Predictor class this handler uses to initiate predictor\n",
+    "                instance if given.\n",
+    "        \"\"\"\n",
+    "        pass\n",
+    "\n",
+    "    def handle(self, request: Request) -> Response:\n",
+    "        \"\"\"Handles a prediction request.\n",
+    "        Args:\n",
+    "            request (Request):\n",
+    "                The request sent to the application.\n",
+    "        Returns:\n",
+    "            The response of the prediction request.\n",
+    "        \"\"\"\n",
+    "        pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then implement a custom `Handler`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile $USER_SRC_DIR/handler.py\n",
+    "\n",
+    "import csv\n",
+    "from io import StringIO\n",
+    "import json\n",
+    "\n",
+    "from fastapi import Response\n",
+    "\n",
+    "from google.cloud.aiplatform.prediction.handler import Handler\n",
+    "\n",
+    "from typing import Optional, Type\n",
+    "\n",
+    "class CprHandler(Handler):\n",
+    "    \"\"\"Default prediction handler for the prediction requests sent to the application.\"\"\"\n",
+    "\n",
+    "    def __init__(\n",
+    "        self, gcs_artifacts_uri, predictor = None,\n",
+    "    ):\n",
+    "        \"\"\"Initializes a Handler instance.\"\"\"\n",
+    "        if predictor is None:\n",
+    "            raise ValueError(\n",
+    "                \"CprHandler must have a predictor class passed to the init function.\"\n",
+    "            )\n",
+    "\n",
+    "        self._predictor = predictor()\n",
+    "        self._predictor.load(gcs_artifacts_uri)\n",
+    "\n",
+    "    async def handle(self, request):\n",
+    "        \"\"\"Handles a prediction request.\"\"\"\n",
+    "        request_body = await request.body()\n",
+    "        data = json.loads(request_body)\n",
```

Comment:
I was thinking the request would not be json encoded. Instead it'll be raw CSV

curl -X POST -H 'Content-Type: text/csv' -d @file.csv [ENDPOINT]

so we'd csv decode the request body itself.

Reply:
Done. Also, changed it to raw-predict.